bifrost/lib/turbolinks/index.ts

```diff
@@ -63,6 +62,8 @@ export const Turbolinks = {
       return new Promise((resolve) => {
         visit.cancelFn = () => resolve();
         visit.renderFn = () => {
+          // store pageContext for restoration visits
+          controller.pageContext = pageContextToCache;
```

Comment:
we now store the new page's pageContext synchronously after snapshot for the old page is cached. 

https://github.com/Alignable/bifrost/blob/4feca985823903be7bafbe83d062b6c68ad23ef2/bifrost/lib/turbolinks/visit.ts#L131-L132